pkg/prometheus/promcfg.go

```diff
@@ -81,14 +85,37 @@ func NewConfigGenerator(logger log.Logger, p monitoringv1.PrometheusInterface, e
 
 	logger = log.WithSuffix(logger, "version", promVersion)
 
+	scrapeClasses, defaultScrapeClass, err := getScrapeClassConfig(cpf)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse scrape classes: %w", err)
+	}
+
 	return &ConfigGenerator{
 		logger:                 logger,
 		version:                version,
 		prom:                   p,
 		endpointSliceSupported: endpointSliceSupported,
+		scrapeClasses:          scrapeClasses,
+		defaultScrapeClass:     defaultScrapeClass,
 	}, nil
 }
 
+func getScrapeClassConfig(cpf monitoringv1.CommonPrometheusFields) (map[string]monitoringv1.ScrapeClass, string, error) {
+	scrapeClasses := map[string]monitoringv1.ScrapeClass{}
```

Comment:
We can start the improvement by pre-allocating the map size
```suggestion
	scrapeClasses := make(map[string]*monitoringv1.ScrapeClass{}, len(cpf.ScrapeClasses))
```

Reply:
Good point 👍🏽